vcenter_operator/phelm.py

```diff
@@ -58,22 +158,39 @@ def delta(self, other):
         # no ordering necessary for delete
         for k in self.items.keys() - other.items.keys():
             delta.actions[k] = 'delete'
-        # sort by (kind, name), so we update ConfigMaps before Deployments, so
-        # that restarting pods can read the new ConfigMaps already
-        for k in sorted(self.items.keys() & other.items.keys(),
-                        key=itemgetter(1, 2)):
+        # sort by kind in order: Secrets, ConfigMaps, Deployments, so we update Secrets before ConfigMaps before
+        # Deployments, so that restarting pods can read the new Secrets and ConfigMaps already
+        for k in (self.items.keys() & other.items.keys()):
             if self.items[k] != other.items[k]:
                 delta.actions[k] = 'update'
                 delta.items[k] = other.items[k]
             # Nothing to do otherwise
-        # sort by (kind, name), so we update ConfigMaps before Deployments, so
-        # that restarting pods can read the new ConfigMaps already
-        for k in sorted(other.items.keys() - self.items.keys(),
-                        key=itemgetter(1, 2)):
+        # sort by kind in order: Secrets, ConfigMaps, Deployments, so we update Secrets before ConfigMaps before
+        # Deployments, so that restarting pods can read the new Secrets and ConfigMaps already
+        for k in (self.items.keys() - other.items.keys()):
             delta.items[k] = other.items[k]
 
+        delta.order_items()
         return delta
 
+    def _sort_resources(self, items):
+        """
+        Sorts the resources in the order of Secrets, ConfigMaps, Deployments.
```

Comment:
That's defined in `RESOURCE_ORDER` above and can change, so we shouldn't write it here. Maybe better "Sort the resources in the order defined by RESOURCE_ORDER".